### PR TITLE
feat(dev/plugin)!: drop support for Node 17 and support Node 20

### PR DESCRIPTION
**Pull request by jwartofsky-yext**
Tested `npm run dev` and `npm run prod` on the starter using Node 20. Page generated without issue.

However, I had to remove references to `import { fetch } from "@yext/pages/util";` from the starter since that no longer exists in pages. With that removed, the starter uses the default fetch function which works fine with Node 18 and newer.

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
         working-directory: ./packages/${{ inputs.package }}
     strategy:
       matrix:
-        node-version: [17.x, 18.x]
+        node-version: [18.x, 19.x, 20.x]
```
Comment (asanehisa) [inline .github/workflows/unit_test.yml:19]: does this one need 19? besides that lgtm

Comment (jwartofsky-yext) [reply .github/workflows/unit_test.yml:19]: Maybe? It had two before and we are still supporting 18, so I thought we would test 18-20

**Review by asanehisa — DISMISSED** [review]

File: .github/workflows/jstest.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: Build with NodeJS 18
         uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version: 20
```
Comment (mkilpatrick) [inline .github/workflows/jstest.yml:24]: Keep this at 18

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        node-version: [17.x, 18.x, 19.x]
+        node-version: [18.x, 19.x, 20.x]
```
Comment (mkilpatrick) [inline .github/workflows/playwright.yml:11]: Remove 19, we only want 18 and 20

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -18,10 +18,10 @@ jobs:
       - name: Install pnpm
         uses: pnpm/action-setup@v2.2.4
 
-      - name: Set node version to 17.x
+      - name: Set node version to 20.x
```
Comment (mkilpatrick) [inline .github/workflows/publish.yml:21]: Do 18 since that's our min version. Everywhere else too.

File: packages/pages/package.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
     }
   },
   "engines": {
-    "node": "^17 || ^18 || ^19"
+    "node": "^18 || ^19 || ^20"
```
Comment (mkilpatrick) [inline packages/pages/package.json:35]: Remove 19

File: packages/pages/src/util/index.ts [hunk 6]
```diff
@@ -1,3 +1,2 @@
-export { fetch } from "./fetch.js";
```
Comment (mkilpatrick) [inline packages/pages/src/util/index.ts:1]: Remove the custom fetch function too

Comment (jwartofsky-yext) [reply packages/pages/src/util/index.ts:1]: Already done

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**